h/services/delete_user.py

```diff
@@ -63,6 +62,11 @@ def _delete_groups(self, groups):
         for group in groups:
             self.request.db.delete(group)
 
+    def _unassign_groups_creator(self, groups):
```

Comment:
I think this is OK here for now as a private method in the delete-user service, but it might make sense to make some of it reusable and stick it on the group-update service, as we're munging extant group records, not deleting 'em. Or even just directly using the group-update's update method to update the creator instead of directly touching the DB here. But that's extra credit and possibly premature optimization :).

In any case, even though this is a private method, mind adding a docstring/comments?

Reply:
I admit this did occur to me but I wasn't sure if that would be setting a new precedent for services calling other services. Just a quick browse around the code, I don't see any services that call into other services and I'm concerned that might make the architecture more complicated. Also note the delete group private method could have called the delete_group service but it didn't and I suspect the reason is to not complicate services with interaction from other services. This is prob something sean would have input on but unfortunately he is out.

Reply:
The pattern is that we pass a specific service method, as needed, to the service in question. Easier to see as example: https://github.com/hypothesis/h/blob/943288b8c3ce9eab36ca9e5570996a2b8d28d829/h/services/group_create.py#L160

No need to change this now but something to consider later.

Reply:
Ah! Ok. Yes perhaps a follow up PR to use that pattern in this service. Seems like there are a couple places where we should do that.

Reply:
I like that plan!